From 1d90727ff2368012d6e2f91b1a3198f626495b7f Mon Sep 17 00:00:00 2001
From: Ming Liu <liu.ming50@gmail.com>
Date: Sat, 25 Jan 2020 14:36:16 +0100
Subject: [PATCH] Replace time.clock() with time.process_time()

The use of time.clock() is deprecated in python 3.8, change to use
time.process_time().

Reference:
https://docs.python.org/3.3/library/time.html#time.clock

Upstream-Status: Pending

Signed-off-by: Ming Liu <liu.ming50@gmail.com>
---
 lib/Crypto/Random/_UserFriendlyRNG.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/Crypto/Random/_UserFriendlyRNG.py b/lib/Crypto/Random/_UserFriendlyRNG.py
index 957e006..d2a0259 100644
--- a/lib/Crypto/Random/_UserFriendlyRNG.py
+++ b/lib/Crypto/Random/_UserFriendlyRNG.py
@@ -74,7 +74,7 @@ class _EntropyCollector(object):
         self._time_es.feed(struct.pack("@I", int(2**30 * (t - floor(t)))))
 
         # Add the fractional part of time.clock()
-        t = time.clock()
+        t = time.process_time()
         self._clock_es.feed(struct.pack("@I", int(2**30 * (t - floor(t)))))
 
 
-- 
2.7.4

